pylasu/model/model.py

```diff
@@ -95,13 +96,56 @@ def provides_nodes(decl_type):
         return isinstance(decl_type, type) and issubclass(decl_type, Node)
 
 
+def get_only_type_arg(decl_type):
+    type_args = get_type_arguments(decl_type)
+    if len(type_args) == 1:
+        return type_args[0]
+    else:
+        return None
+
+
+def process_annotated_property(name, decl_type, known_property_names):
+    multiplicity = Multiplicity.SINGULAR
+    is_reference = False
+    if get_type_origin(decl_type) is ReferenceByName:
```

Comment:
In Python I often get confused by `is` because I should use `isinstance` instead. I am not sure if that is the case here, but I mention it to be on the safe side